Deeploy/TilingExtension/CodeTransformationPasses/TilingPrototypes.py

```diff
@@ -340,10 +352,14 @@ def generateSetupAndTeardownCode(cls, executionBlock: ExecutionBlock, metaInfo:
             "tileIdx": 0
         })
 
-        for measurementName in [
-                "kernel_start", "kernel_end", "ingress_dma_wait_start", "ingress_dma_wait_end", "egress_dma_wait_start",
-                "egress_dma_wait_end"
-        ]:
+        measurementNameList = [
+            "ingress_dma_wait_start", "ingress_dma_wait_end", "egress_dma_wait_start", "egress_dma_wait_end"
+        ]
+
+        if metaInfo.kernelLevelTiling:
+            measurementNameList = ["kernel_start", "kernel_end"] + measurementNameList
+
+        for measurementName in measurementNameList:
```

Comment:
Looks correct but there is quite some code duplication with `ProfilingSingleBufferingTilingMixIn`. Can this be optimized? Would make it easier in case there are future changes.